File: examples/AppModel/booksandrecords-spring-boot/src/main/scala/com/ms/booksandrecords/commands/OpenDeal.scala
package com.ms.booksandrecords.commands

import com.fasterxml.jackson.annotation.JsonProperty
import org.springframework.stereotype.Component

@Component class OpenDeal() extends DealCmd {
  private var id = ""
  private var product = ""
  private var price = .0f
  private var quantity = 0

  def this(@JsonProperty("id") id: String, @JsonProperty("productid") productId: String, @JsonProperty("price") price: Float, @JsonProperty("quantity") quantity: Int) {
    this()
    this.id = id
    product = productId
    this.price = price
    this.quantity = quantity
  }

  def getId = id

  def setId(id: String) = this.id = id

  def getProduct = product

  def setProduct(product: String) = this.product = product

  def getPrice = price

  def setPrice(price: Float) = this.price = price

  def getQuantity = quantity

  def setQuantity(quantity: Int) = this.quantity = quantity
}

File: examples/AppModel/booksandrecords-spring-boot/src/main/java/com/ms/booksandrecords/commands/OpenDeal.java
package com.ms.booksandrecords.commands;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.springframework.stereotype.Component;


@Component
public class OpenDeal extends DealCmd {
    private String id, product;
    private float price;
    private int quantity;

    public OpenDeal(){ }
    public OpenDeal(@JsonProperty("id") String id, @JsonProperty("productid") String productId,
                    @JsonProperty("price") float price, @JsonProperty("quantity") int quantity) {
        this.id = id;
        product = productId;
        this.price = price;
        this.quantity = quantity;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getProduct() {
        return product;
    }

    public void setProduct(String product) {
        this.product = product;
    }

    public float getPrice() {
        return price;
    }

    public void setPrice(float price) {
        this.price = price;
    }

    public int getQuantity() {
        return quantity;
    }

    public void setQuantity(int quantity) {
        this.quantity = quantity;
    }
}
